cmd/snap-confine/snap-confine.apparmor.in

```diff
@@ -550,22 +552,33 @@
         /tmp/.snap/{,**} rw,
         # Allow mounting/unmounting any part of $SNAP over to a temporary place
         # in /tmp/.snap/ during the preparation of a writable mimic.
-        mount options=(bind, rw) /snap/*/** -> /tmp/.snap/**,
+        mount options=(bind, rw) /*/** -> /tmp/.snap/**,
         # Allow mounting tmpfs over the original $SNAP/** directory.
-        mount fstype=tmpfs options=(rw) tmpfs -> /snap/*/**,
+        mount fstype=tmpfs options=(rw) tmpfs -> /*/**,
         # Allow bind mounting anything from the temporary place in /tmp/.snap/
         # back to $SNAP/** (to re-construct the data that was there before).
-        mount options=(bind, rw) /tmp/.snap/** -> /snap/*/**,
+        mount options=(bind, rw) /tmp/.snap/** -> /*/**,
         # Allow unmounting the temporary directory in /tmp once it is no longer
         # necessary.
         umount /tmp/.snap/**,
 
-        # Allow creating missing directories under $SNAP once the writable
-        # mimic is ready.
+        # Allow creating missing directories anywhere under the root directory
+        # (but not in the root directory itself) where they need to be created
+        # as a mount point for layouts or for content sharing. This is a
+        # superset of other cases so they are removed
         / r,
-        /snap/ r,
-        /snap/{,**} r,
-        /snap/*/** w,
+        /** r,
+        /*/** w,
```

Comment:
```
/{,var/}snap/{,**} r,
/{,var/}snap/*/** w,
```
?

Reply:
No, this is actually correct I'm afraid. Otherwise we cannot traverse the filesystem enough to create new directories. Simple example: `mount --bind $SNAP/foo /usr/share/foo`

Reply:
Wow. Being able to write everywhere on the disk. Hmm. This means it could rewrite its own policy, overwrite itself, scribble across /dev/.... There is really no point in having a profile if we allow writing (nearly) everywhere on the disk....

I'm not sure how to proceed. `/*/**/ w,` would almost be ok, but still not fantastic, but that won't work for symlinks and files....

Grasping at straws here.... Maybe this is the point where we have per-snap snap-update-ns profiles?

Reply:
Or deny rules but ... yeah not perfect.

Reply:
Blocker: please add a `# FIXME: update this with per-snap snap-update-ns profiles`.